ibm/service/database/validators.go

```diff
@@ -0,0 +1,137 @@
+// Copyright IBM Corp. 2017, 2022 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package database
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/IBM/cloud-databases-go-sdk/clouddatabasesv5"
+)
+
+/* TODO move other validators in here */
+
+/* VERSION VALIDATOR */
+
+type Version struct {
+	Version     string
+	Type        string
+	Status      string
+	IsPreferred bool
+	Transitions []VersionTransition
+}
+
+type VersionTransition struct {
+	Application         string
+	Method              string
+	FromVersion         string
+	ToVersion           string
+	SkipBackupSupported *bool
+}
+
+func expandVersion(version clouddatabasesv5.VersionsCapabilityItem) *Version {
+	expandedVersion := &Version{
+		Version:     *version.Version,
+		Status:      *version.Status,
+		IsPreferred: *version.IsPreferred,
+	}
+
+	if version.Transitions != nil {
+		var transitions []VersionTransition
+
+		for _, transition := range version.Transitions {
+			versionTransition := VersionTransition{
+				Application: *transition.Application,
+				Method:      *transition.Method,
+				FromVersion: *transition.FromVersion,
+				ToVersion:   *transition.ToVersion,
+			}
+
+			if skipBackup := transition.SkipBackupSupported; skipBackup != nil {
+				versionTransition.SkipBackupSupported = skipBackup
+			}
+
+			transitions = append(transitions, versionTransition)
+		}
+		expandedVersion.Transitions = transitions
+	}
+
+	return expandedVersion
+}
+
+func (v *Version) isVersionUpgradeAllowed(version string) bool {
+	for _, transition := range v.Transitions {
+		if transition.ToVersion == version && transition.Method == "in-place" {
+			return true
+		}
+	}
+	return false
+}
+
+func (v *Version) isSkipBackupUpgradeAllowed(version string) bool {
+	for _, transition := range v.Transitions {
+		if transition.ToVersion == version && transition.SkipBackupSupported != nil && transition.Method == "in-place" {
+			return *transition.SkipBackupSupported
+		}
+	}
+	return false
+}
+
+func (v *Version) hasUpgradeVersions() bool {
+	for _, transition := range v.Transitions {
+		if transition.Method == "in-place" {
+			return true
+		}
+	}
+	return false
+}
+
+func (v *Version) getAllowedVersionsList() []string {
+	var allowedList []string
+	for _, transition := range v.Transitions {
+		if transition.Method == "in-place" {
+			allowedList = append(allowedList, transition.ToVersion)
+		}
+	}
+	return allowedList
+}
+
+var fetchDeploymentVersionFn = fetchDeploymentVersion
```

Comment:
Is this reassignment required?

Reply:
It is so we can test the function. This was the only way I could find to do that. Please let me know if there is another way to test?

Reply:
You might be able to move that line to the test file, as long as they are in the same package